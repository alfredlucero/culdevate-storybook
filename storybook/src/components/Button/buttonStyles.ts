import { css } from 'styled-components';

const buttonStyle = css`
  background-color: #fff;
  border: none;
  cursor: pointer;
`;

export { buttonStyle };
