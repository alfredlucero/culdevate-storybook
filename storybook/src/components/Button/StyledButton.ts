import styled, { css } from 'styled-components';

import { buttonStyle } from './buttonStyles';

export interface StyledButtonProps {
  type: 'primary' | 'secondary' | 'danger' | 'warning' | 'plaintext';
  disabled?: boolean;
  round?: boolean;
  roundColor?: string;
}

const StyledButton = styled.button`
  ${buttonStyle}

  ${(props:StyledButtonProps) => props.round && css`
    position: relative;
    overflow: hidden;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 50px;
    height: 50px;
    font-weight: bold;
    font-size: 2rem;
    padding: 1rem;
    border-radius: 100%;
    border: 2px solid #eee;
    color: ${props.roundColor};
    background-color: #fff;
    cursor: pointer;
    z-index: 1;
    transition: 0.15s ease-in;


    &:before {
      content: "";
      position: absolute;
      background-color: ${props.roundColor};
      bottom: 0;
      left: 0;
      right: 0;
      top: 100%;
      z-index: -1;
      transition: top 0.2s ease-in;
    }

    &:hover {
      border: 2px solid ${props.roundColor};
      background-color: ${props.roundColor};
      color: #fff;
    }

    &:hover:before {
      top: 0;
    }
  `}
`;

export { StyledButton };
