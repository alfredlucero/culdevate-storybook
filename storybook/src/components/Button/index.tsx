import * as React from 'react';

import styled from 'styled-components';
import { StyledButton } from './StyledButton';

export interface ButtonProps {
  type: 'primary' | 'secondary' | 'danger' | 'warning' | 'plaintext' | 'round';
  disabled?: boolean;
  roundColor?: string;
  children?: React.ReactElement<any> | string;
  onClick?: (event: any) => void;
}

const Button: React.SFC<ButtonProps> = ({
  children,
  ...styledButtonProps
}) => {

  return (
    <StyledButton {...styledButtonProps}>
      {children}
    </StyledButton>
  )
};

export default Button;
