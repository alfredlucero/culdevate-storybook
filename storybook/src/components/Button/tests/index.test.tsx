import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from '../index';

describe('<Button />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button type="plaintext" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render its children', () => {
    const children = 'Sample button text';
    const renderedComponent = shallow(
      <Button type="plaintext">{children}</Button>
    );
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
