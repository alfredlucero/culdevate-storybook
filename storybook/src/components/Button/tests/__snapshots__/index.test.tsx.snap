// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
.c0 {
  background-color: #fff;
  border: none;
  cursor: pointer;
}

<button
  className="c0"
  type="plaintext"
/>
`;

exports[`<H1 /> should render correctly 1`] = `
.c0 {
  color: #595959;
  font-size: 2rem;
  font-family: 'Inconsolata';
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

<h1
  className="c0"
/>
`;
